core/ExecutePipe.cpp

```diff
@@ -170,6 +185,17 @@ namespace olympia
             scoreboard_views_[reg_file_]->setReady(dest_bits);
         }
 
+        // Testing mode to inject random branch misprediction to stress flushing mechanism
+        if (enable_random_misprediction_)
+        {
+            if (ex_inst->isBranch() && (std::rand() % 20) == 0)
```

Comment:
This is awesome, but can a failure be reproduced?  Probably should add a parameter to set a seed...

Also, can the 20 be parameterized as well?  To increase the chances (for interesting experiments).

Reply:
You could change the parameter `enable_random_misprediction` to `random_misprediction_rate` and default it to zero.

Reply:
I forgot to reply back to this last night.

With std::rand, if the seed isn't set then it's always seeded with 1 according to google so should be reproducible (for now)

I think the random_misprediction_rate is a good idea. Something like tuning mispredictions per 1k branches.
I don't have a good knowledge of the C++ random libraries, but I saw I could use the discrete_distribution class. Happy to do this in another PR, as I think it'll need some back and forth discussion.